Makefile

```diff
@@ -0,0 +1,13 @@
+GOBIN=$(shell pwd)/bin
+
+install-lint:
+	@if [ ! -f "$(GOBIN)/golangci-lint" ]; then \
+		curl -sSfL -q https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GOBIN) v1.42.0; \
```

Comment:
I'm personally not a fan of just running shell scripts to install tooling like this, it makes more sense to install it in a more package manager friend way. (Maybe just point people to the `golangci-lint` project instead where they have platform specific instructions on how to properly install it.

Reply:
I reason I used it this way to install was that it covers `Linux`, `Mac` and to some extent `Windows` (using Git Bash or WSL). I wanted to install it using `go get`, but that method is not recommended.
```
Note: such go get installation aren't guaranteed to work. We recommend using binary installation.
```

Reply:
Understood, could you add a conditional so that on macOS it strictly uses `brew` and not the install script? (`brew install golangci-lint`)

Reply:
The way `make lint` currently works is that it installs `golangci-lint` inside `bin` directory of this repo. If I change it to use `brew` to install the linter, it will install `golangci-lint` on developer's machine. I don't think it a good strategy to install software on developer's machine, that's only relevant to this repo.

Reply:
what about tools.go?   https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module